src/overlays/actors/ovl_En_Dnq/z_en_dnq.c

```diff
@@ -22,55 +52,426 @@ const ActorInit En_Dnq_InitVars = {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_KINGNUTS_DAMAGE);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A52B68.s")
+        if (((this->unk_398 == 3) || (this->unk_398 == 6)) &&
+            func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+            func_80A5257C(this, this->unk_398 + 1);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A52C6C.s")
+        func_800EDF24(&this->actor, globalCtx, temp_v0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A52CF8.s")
+void EnDnq_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnq* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A52D44.s")
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 14.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600EB48, NULL, this->jointTable, this->morphTable, 33);
+    this->unk_398 = -1;
+    func_80A5257C(this, 0);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+    Actor_SetScale(&this->actor, 0.02f);
+    this->actor.targetMode = 1;
+    this->unk_386 = 0;
+    this->unk_37C = 0;
+    func_8013AED4(&this->unk_37C, 3, 7);
+    if (gSaveContext.weekEventReg[9] & 0x80) {
+        this->unk_3A4 = 1;
+    } else {
+        this->unk_3A4 = 0;
+    }
+    this->unk_144 = func_80A52530;
+    this->actionFunc = func_80A52DC8;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A52DC8.s")
+void EnDnq_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnq* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A52FB8.s")
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/func_80A53038.s")
+void EnDnq_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDnq* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/EnDnq_Init.s")
+    if (!func_80A52D44(this, globalCtx) && func_80A52648(this, globalCtx)) {
+        func_80A53038(this, globalCtx);
+        SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dnq/EnDnq_Destroy.s")
+    this->actionFunc(this, globalCtx);
+    func_80A52B68(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 30.0f, 12.0f, 0.0f, 4);
+    this->unk_394 = this->actor.xzDistToPlayer;
+    func_80A52C6C(this, globalCtx);
+    func_8013C964(&this->actor, globalCtx, this->unk_390, fabsf(this->actor.yDistToPlayer) + 1.0f, 0,
+                  this->unk_37C & 7);
+    this->actor.xzDistToPlayer = this->unk_394;
+    Actor_SetHeight(&this->actor, 46.0f);
+    func_80A52604(this, globalCtx);
```

Comment:
You could put this in an `else` and remove the `return` in the first condition.

Reply:
I think it looks better with a return, but I'll change it.